### PR TITLE
v5.0.0 release

### PR DESCRIPTION
**Pull request by marandaneto**
PR for v5.0.0

_#skip-changelog_

### CODE REVIEW

**Comment by krystofwoldrich** [comment]
I've tried RN 0.70 with the latest alpha release and everything seems to be working.

To my surprise activating the new TurboModules didn't break the build. I haven't find anywhere if you can mix NativeModules and TurboModules, but it would make sense that you can otherwise the transition to the new architecture would be too painful.

I also tried the new Fabric renderer and errors being reported as expected.

**Comment by krystofwoldrich** [comment]
The perf ios test is failing because of the cocoa pods codesigning from Xcode 14
- https://github.com/facebook/react-native/issues/34673

**Comment by krystofwoldrich** [comment]
Can I get a final review on this PR? 

It's a lot of changes and the individual changes have been reviewed already. We just need to skim it, just in case something slipped thru.

**Review by krystofwoldrich — APPROVED** [review]
LGTM 🚀 

But I'll wait for more people to skim thru this before merging. (All individual changes have been already reviewed.)

This will be merged without squash as we would lose too much information about the changes here.

File: .github/workflows/codegen.yml
Comment (kahest) [inline .github/workflows/codegen.yml:15]: what is the context of this fix? is this blocked?

File: src/js/version.ts
Comment (kahest) [inline src/js/version.ts:3]: Should be `5.0.0` now?

File: .github/workflows/codegen.yml
Comment (krystofwoldrich) [reply .github/workflows/codegen.yml:15]: Not blocked anymore, the fix was released in [RN 0.70.4](https://github.com/facebook/react-native/releases/tag/v0.70.4).

File: src/js/version.ts
Comment (krystofwoldrich) [reply src/js/version.ts:3]: The release CI pipeline will update the version.

**Comment by chrisedington** [comment]
Looking forward to this for 0.69 :)

**Pull request merged**
